### PR TITLE
Fix #1215 Remove incorrect references to non-existent MagicDraw items

### LINKED ISSUE #1215: Incorrect references to non-existent MagicDraw files
| | |
|:---|:---|
|**_F´ Version_**| 3.0.0 |
|**_Affected Component_**| Documentation |
---
## Problem Description

The next two issues, #1166 #974, required a project purge to remove the now obsolete files, including the mdxml.

Since the @LeStarch commit https://github.com/nasa/fprime/pull/1167/commits/97de3c264553a5348ecf6c26d9e30cdc95aa2567, the last traces of MagicDraw support have been "officially" effective, including the removal of the mdxml files.

However, some component ``README`` files still contain references to MagicDraw project files.

Exemple:
https://github.com/nasa/fprime/blob/313ef0556baec4a981f03e66de64d9193a84b867/Fw/Cmd/README#L7

## How to Reproduce

1. Launch ``grep -nr 'mdxml'`` inside fprime folder
2. See the following output:

```ps
fprime$ grep -nr 'mdxml' 
.dockerignore:4:**.mdxml
Binary file .git/objects/pack/pack-ba5f57e658c47e79b316f15efb42028e0016b801.pack matches
.github/actions/spelling/excludes.txt:21:\.mdxml$
.github/actions/spelling/expect.txt:935:mdxml
.gitignore:36:*.mdxml.bak
docs/UsersGuide/dev/magicdraw.md:18:or examples will work.  Open up the `Ref/Top/REFApplication.mdxml` file to begin. If import errors arise, the user will
Fw/Cmd/README:7:CmdModule.mdxml - MagicDraw project file describing command interface
Fw/Com/README:4:ComModule.mdxml - A MagicDraw project file describing a communication buffer port
Fw/Log/README:8:LogModule.mdxml - MagicDraw project file describing event interface
Fw/Prm/README:7:PrmModule.mdxml - MagicDraw project file that describes the parameter interface
Fw/Time/README:5:TimeModule.mdxml - MagicDraw project file describing time interface
Fw/Tlm/README:7:TlmModule.mdxml - MagicDraw project file for telemetry interface
Svc/ActiveLogger/README:7:ActiveLoggerModule.mdxml - MagicDraw project file describing the active logger
Svc/ActiveRateGroup/README:7:ActiveRateGroupModule.mdxml - The MagicDraw project file describing the component
Svc/CmdDispatcher/README:7:CmdDispatcherModule.mdxml - MagicDraw project file describing component
Svc/PassiveConsoleTextLogger/README:5:PassiveTextLoggerModule.mdxml - MagicDraw project file that describes the component
Svc/PolyDb/README:7:PolyDbModule.mdxml - MagicDraw project file that describes the PolyDb component
Svc/PolyIf/README:5:PolyIfModule.mdxml - MagicDraw project file that describes the interface
Svc/PrmDb/README:10:PrmDbModule.mdxml - MagicDraw project file that describes the parameter database component
Svc/RateGroupDriver/README:7:RateGroupDriverModule.mdxml - MagicDraw project file that describes rate group driver component
Svc/Sched/README:4:SchedModule.mdxml - MagicDraw project file that describes scheduler port
Svc/Time/README:4:TimeCompModule.mdxml - The MagicDraw project file that describes the time component.

```

## Expected Behavior

Do not provide misleading or incorrect information about non-existent files to the user when browsing the repository or reading the component documentation.

---

🏷️ Advised issue label: ``Documentation``  ``Easy First Issue``.

### PR DESCRIPTION
**Pull request by ThibFrgsGmz**
| | |
|:---|:---|
|**_Originating Project/Creator_**| @ThibFrgsGmz |
|**_Affected Component_**| Documentation / README  |
|**_Affected Architectures(s)_**|  N/A|
|**_Related Issue(s)_**| #1215  |
|**_Has Unit Tests (y/n)_**| N/A |
|**_Builds Without Errors (y/n)_**|  N/A |
|**_Unit Tests Pass (y/n)_**| N/A |
|**_Documentation Included (y/n)_**| y |

---
## Change Description

This PR aims to fix issue #1215 where some component README files still contain references to MagicDraw project files.

## Rationale

See #1215

## Testing/Review Recommendations

N/A

## Future Work

An analysis must be done to update or delete the READMEs of the components and integrate them in the SDD.md for example.

### CODE REVIEW

File: Fw/Cmd/README [hunk 1]
```diff
@@ -3,5 +3,4 @@ CmdResponseAi.xml - XML definition of command response port, which is used to re
 CmdRegPortAi.xml - XML definition of a command registration port, which is used by components to register their set of commands
 CmdArgBuffer.hpp(.cpp) - C++ Definition of command argument buffer. Buffer is used to store serialized arguments for component commands
 CmdString.hpp(.cpp) - C++ Definition of a command string argument. Used by the code generator when a command has a string argument
-CmdPacket.hpp(.cpp) - C++ Definition of command packet. It is a derived class of ComPacket, which is the type used to send and receive data from ground or test software
-CmdModule.mdxml - MagicDraw project file describing command interface
\ No newline at end of file
+CmdPacket.hpp(.cpp) - C++ Definition of command packet. It is a derived class of ComPacket, which is the type used to send and receive data from ground or test software
```
Comment (ThibFrgsGmz) [inline Fw/Cmd/README:6]: @LeStarch Do I have to add a new line at the end of each file?

**Comment by ThibFrgsGmz** [comment]
@LeStarch 
Shall we remove the [MagicDraw](https://github.com/nasa/fprime/blob/313ef0556baec4a981f03e66de64d9193a84b867/docs/UsersGuide/dev/magicdraw.md) user guide  since the MagicDraw files are no longer present in the repo. So the user theoretically cannot follow the described procedure.

**Comment by Joshua-Anderson** [comment]
> Shall we remove the MagicDraw user guide since the MagicDraw files are no longer present in the repo. So the user theoretically cannot follow the described procedure.

Yeah, magicdraw is no longer supported so I agree that there's no need to keep the user guide. Users can always reference older versions of the documentation if they need it.

File: Fw/Cmd/README (hunk 1 above)
Comment (LeStarch) [reply Fw/Cmd/README:6]: No, you should not need to.

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
@ThibFrgsGmz @Joshua-Anderson  is correct.  It exists in the documentation for versions which have MagicDraw support, so lets remove it.  Also make sure to remove it's link from the documentation table of contents.

**Comment by ThibFrgsGmz** [comment]
Okay, I'll do it. I'll change the status from WIP to Pending when I'm ready for the merge. I don't have the ability to assign labels to my Issues/PRs myself, which is why I put "current status: WIP" at the end of my PR.
Otherwise, feel free to commit to my branch if you wish.

**Comment by ThibFrgsGmz** [comment]
I wonder if it wouldn't be better to remove the READMEs from the components, or just the list of component files.
The list doesn't seem to be up to date for all components (I haven't analyzed all the lists in detail) so it seems to me that it's a pretty big maintenance burden to keep these lists.

What do you think about it?

**Comment by LeStarch** [comment]
I agree! Our SDDs should contain the documentation for a component, and we haven't touched these READMEs in a long time.

**Comment by ThibFrgsGmz** [comment]
Ok, I conclude that all the component readme's should be removed because of the associated maintenance burden. For example, not all of them are present in the components, nor do they list all the component files.

Should we do this in this PR or open another one once this one is completed/accepted?

**Comment by LeStarch** [comment]
@ThibFrgsGmz I'd do it here since you'd still need to remove the .mdxml lines.

**Comment by ThibFrgsGmz** [comment]
It seems too decilent to me to remove all READMEs from components in this PR. Indeed, some of them contain a lot of information and without a deep analysis, I am not able to say if they are deprecated.

For example :

- [Svc/ComLogger/README.md](https://github.com/nasa/fprime/blob/devel/Svc/ComLogger/README.md)
- [Svc/ComSplitter/README.md](https://github.com/nasa/fprime/blob/devel/Svc/ComSplitter/README.md)

**Comment by ThibFrgsGmz** [comment]
I think I'm done with the PR. I changed the status to Pending for review.

**Review by LeStarch — APPROVED** [review]

**Pull request merged**